### PR TITLE
[Heartbeat] Populate url.port field for http(s) fields.

### PR DESCRIPTION
**Pull request by andrewvc**
The url.* ECS fields are used for all sorts of stuff. The golang `url` package doesn't infer the URL for URLs with the http(s) schemes, which means that there will be no port value for url.port for these common URLs. This patch explicitly fills in the value in these situations.

This change also fixes some other tests by using an explicit port for TCP which is required in heartbeat. Additionally, it standardizes on the `uint16` type for port values which is common in the go ecosystem.

No Changelog necessary since this is for 7.0 only and does not affect 6.x

### CODE REVIEW

**Comment by andrewkroh** [comment]
@webmat I think ECS isn't really clear about what to do in case the port isn't included in the raw URL. I would like to have clarity in the ECS spec for this case. Should we populate `url.port` if a port wasn't present in the URL but we can infer the port based on the scheme? Or should we strictly populate the `url.*` fields based on what's in the raw URL? I can make arguments on both side.

- It might be some kind of anomaly or automated process if ":80" or ":443" is included in the raw URL from the user. So searching for `url.scheme:http AND _exists_:url.port` might find interesting things.

- Having `url.port` always populated makes exploring the data and filtering a lot easier.

**Comment by andrewvc** [comment]
Yeah, I agree it shouldn't be required, but I think including should be a strong recommendation if that's a thing. Aggregations are really confusing when there's nothing under port 80.

File: heartbeat/monitors/wrappers/util.go [hunk 1]
```diff
@@ -58,7 +58,15 @@ func URLFields(u *url.URL) common.MapStr {
...
 	if u.Port() != "" {
-		fields["port"], _ = strconv.ParseUint(u.Port(), 10, 16)
+		// Returns a uint64 believe it or not.
+		val, _ := strconv.ParseUint(u.Port(), 10, 16)
+		fields["port"] = uint16(val)
+	} else {
+		if u.Scheme == "http" {
+			fields["port"] = uint16(80)
+		} else if u.Scheme == "https" {
+			fields["port"] = uint16(443)
```
Comment (ruflin) [inline heartbeat/monitors/wrappers/util.go]: Should we have an ending "else" to catch edge cases we were not thinking of?

Comment (andrewvc) [reply heartbeat/monitors/wrappers/util.go]: Good question. I believe the answer is no. 

ICMP does not have a port
TCP has no notion of a default port, and our configs require one to be set anyway.

**Pull request merged**